appcues/src/main/java/com/appcues/ui/AppcuesOverlayViewManager.kt

```diff
@@ -0,0 +1,93 @@
+package com.appcues.ui
+
+import android.app.Activity
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.ComposeView
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.findViewTreeLifecycleOwner
+import com.appcues.R
+import com.appcues.databinding.AppcuesOverlayLayoutBinding
+import com.appcues.logging.Logcues
+import com.appcues.monitor.AppcuesActivityMonitor
+import com.appcues.ui.composables.AppcuesComposition
+import com.appcues.ui.primitive.EmbedChromeClient
+import com.appcues.util.getNavigationBarHeight
+import com.appcues.util.getStatusBarHeight
+import org.koin.core.scope.Scope
+
+class AppcuesOverlayViewManager(private val scope: Scope) : DefaultLifecycleObserver {
+
+    private val logcues: Logcues by lazy { scope.get() }
+
+    private var viewModel: AppcuesViewModel? = null
+
+    private var shakeGestureListener: ShakeGestureListener? = null
+
+    override fun onResume(owner: LifecycleOwner) {
+        super.onResume(owner)
+        viewModel?.onResume()
+        shakeGestureListener?.start()
+    }
+
+    override fun onPause(owner: LifecycleOwner) {
+        super.onPause(owner)
+        viewModel?.onPause()
+        shakeGestureListener?.stop()
+    }
+
+    fun addView() {
+        AppcuesActivityMonitor.activity?.run {
```

Comment:
so now that we have the Activity reference always available, are there other parts of the code we can simplify (in later passes is fine) - like the Debugger, I assume? We could update the public API in Appcues.kt to no longer require passing the Activity reference in on the `debug()` call?

Reply:
correct, we can make that call change if we move foward with this PR